python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -279,6 +345,24 @@ async def setup_laser(self, mode: LaserDetailedState) -> None:
                 f"{mode} not an acceptable LaserDetailedState [CONTINOUS, BURST, TRIGGER]"
             )
 
+        await self.change_laser_optical_configuration(optical_configuration)
+        await self.change_laser_wavelength(wavelength, use_projector)
+
+    async def laser_start_propagate(self) -> None:
+        """Start the propagation of the Tunable Laser"""
+        # get laser detailed state
+
+        await self.rem.tunablelaser.cmd_startPropagateLaser.start(
+            timeout=self.laser_warmup
+        )
+
+    async def laser_stop_propagate(self) -> None:
+        """Stop the propagation of the Tunable Laser"""
+
+        await self.rem.tunablelaser.cmd_stopPropagateLaser.start(
+            timeout=self.long_timeout
+        )
+
```

Comment:
Same thing here, you should add a check that the detailed state is nonpropagating before this method completes.